src/uu/chmod/src/chmod.rs

```diff
@@ -35,14 +36,69 @@ mod options {
     pub const FILE: &str = "FILE";
 }
 
-#[uucore::main]
-pub fn uumain(args: impl uucore::Args) -> UResult<()> {
-    let mut args = args.collect_lossy();
+// "-w -w" is the same as "-w,-w"
+// "-w file -w" is the same as "-w,-w file"
+// "-- -w file" is the same as "-w file"
+fn extract_negative_modes(mut args: impl uucore::Args) -> (Option<String>, Vec<OsString>) {
+    // we look up the args until "--" is found
+    // "-mode" will be extracted into parsed_cmode_vec
+    let mut parsed_cmode_vec = Vec::new();
+    let pre_double_hyphen_args: Vec<OsString> = args
```

Comment:
It's probably fine like this, but you could look into [`Iterator::partition`](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.partition) for splitting the args into `parsed_cmod_vec` and `pre_double_hyphen_args`.

Reply:
Definitely check that. I thought about it but it wasn't possible for some reason but I guess the current state allows me to use it :)